lib/mongo/uri.rb

```diff
@@ -636,6 +740,11 @@ def ms_convert(value)
     def hash_extractor(value)
       value.split(',').reduce({}) do |set, tag|
         k, v = tag.split(':')
+        if v.nil?
+          log_warn("#{value} is not a valid URI hash")
```

Comment:
Specify the option that the value goes with.

Reply:
Fixed